src/Microsoft.AspNetCore.HttpOverrides/IPNetwork.cs

```diff
@@ -1,12 +1,245 @@
+                return true;
+            }
+            else
+            {
+                length = 0;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Determines whether a string is a valid IP network.
+        /// </summary>
+        /// <remarks>
+        /// Prefix length is optional. Defaults to /32 for IPv4 and /128 for IPv6.
+        /// </remarks>
+        /// <example>
+        /// IPNetwork ipn;
+        /// if (IPNetwork.TryParse("198.51.100.0/27", out ipn))
+        ///     Console.WriteLine("prefix/length = {0}/{1}", ipn.Prefix, ipn.PrefixLength);
+        /// else
+        ///     Console.WriteLine("Invalid network string");
+        /// </example>
+        /// <param name="networkString">A network string</param>
+        /// <param name="network">An Microsoft.AspNetCore.HttpOverrides.IPNetwork instance if networkString is valid; otherwise null.</param>
+        /// <returns>true if networkString is a valid IP address; otherwise, false.</returns>
+        public static bool TryParse(string networkString, out IPNetwork network)
+        {
+            if (networkString == null)
+            {
+                network = null;
+                return false;
+            }
+
+            IPAddress prefix;
+            byte length;
+
+            var parts = networkString.Split(PrefixLengthSeparators, 2);
+
+            if (IPAddress.TryParse(parts[0], out prefix) && TryParsePrefixLength(prefix, parts, out length))
+            {
+                network = new IPNetwork(prefix, length);
+                return true;
+            }
+            else
+            {
+                network = null;
+                return false;
+            }
+
+        }
+
+        /// <summary>
+        /// Converts an IP network string to an Microsoft.AspNetCore.HttpOverrides.IPNetwork instance.
+        /// </summary>
+        /// <remarks>
+        /// Prefix length is optional. Defaults to /32 for IPv4 and /128 for IPv6.
+        /// </remarks>
+        /// <example>
+        /// IPNetwork.Parse("198.51.100.0/27");
+        /// IPNetwork.Parse("2001:db8::/64");
+        /// </example>
+        /// <param name="networkString">A network string</param>
+        /// <returns>An Microsoft.AspNetCore.HttpOverrides.IPNetwork instance</returns>
+        public static IPNetwork Parse(string networkString)
+        {
+            if (networkString == null)
+            {
+                throw new ArgumentNullException(nameof(networkString));
+            }
+
+            var parts = networkString.Split(PrefixLengthSeparators, 2);
+
+            var prefix = IPAddress.Parse(parts[0]);
+
+            byte length;
+            if (!TryParsePrefixLength(prefix, parts, out length))
+            {
+                throw new FormatException("Invalid prefix length");
+            }
+
+            return new IPNetwork(prefix, length);
+        }
+
+        public IPNetwork(IPAddress prefix)
```

Comment:
This constructor doesn't make sense for use with ForwardedHeadersOptions (and you never call it). Single addresses go on KnownProxies.